### ISSUE TITLE
DAC breakout board

### ISSUE THREAD

**Issue by ijnekenamay** [joeSeggiola/arduino-eurorack-projects#5]
I apologize if I have already finished the discussion somewhere else.
Regarding the DAC breakout board, it seems that GY-MCP4728 is already not available anywhere.
There is an Adafruit breakout board on the market instead, do you think a simple replacement would work?
Thanks in advance.
https://www.adafruit.com/product/4470

**Comment by joeSeggiola** [comment]
Yeah, unfortunately the GY-MCP4728 breakout boards I was using are quite impossible to find now at reasonable prices.

The Adafruit breakout board you found should work, it has the same exact IC on it. The pin-out is compatible, but the layout (footprint) is not, so you won't be able to solder it directly on the PCB I published. You have to use some cables instead. Or maybe you are not using the PCB, so it won't an issue at all for you. Let me know if you manage to do it, or if you need any help! I’m here.

If you're using a couple of them for **In CV**, remember to [set addresses](https://github.com/joeSeggiola/arduino-eurorack-projects/tree/master/in-cv#schematic) before soldering. If it's one for **MIDI 4+1**, it should be come already set with default address #0.

A different option, if you’re comfortable soldering SMD, could be make a compatible breakout board yourself, using bare SMT MCP4728. Adafruit publishes a schematic for theirs [here](https://learn.adafruit.com/assets/85938).

**Comment by ijnekenamay** [comment]
Thank you!
 I'll try then.

**Comment by ijnekenamay** [comment]
Hi, I have fabricated a pcb since then and started build the MIDI 4+1, the DAC is an MCP4728 breakout board from Adafruit, connected with ribbon cables.
https://www.adafruit.com/product/4470

However, I am having trouble with v/oct not working in any of them.
I am trying to check the connections and check the code as I am getting the GATE correctly and I think the DAC is the culprit.
Is there anything you can tell me in advance?
Thanks.

PS.
I have confirmed that the DAC is producing a constant voltage. It may be that the calibration process is not working properly.
The CV output always points to 5V and does not respond to MIDI keyboard note input. Thus, the calibration process could not be performed correctly.

**Comment by ijnekenamay** [comment]
Hey! The newly purchased Adifruit_MCP4728 has an MCP4728A4 to be exact. The address of this IC is 0x64 and the midi4plus1 code needs to be corrected.
I have thus successfully completed the calibration process!

Thanks.

**Comment by joeSeggiola** [comment]
Thank you @ijnekenamay for your valuable feedback! I'll leave the issue open to remind myself to add a note on the read-me page and BOM, since the DAC breakout I originally used is now impossible to find.

**Comment by ijnekenamay** [comment]
You may have already noticed this, but I will include the edited part just in case.
https://github.com/joeSeggiola/arduino-eurorack-projects/blob/master/midi4plus1/midi4plus1.ino

Change the number in line 158 to 4 in
dac.init(Wire, 0);
↓
dac.init(Wire, 4);.

Please edit the documentation.
Thanks.

**Comment by b2unit69** [comment]
I just wonder...I'm using adafruit mcp4728.

Is C4 the standard for raising and lowering the voltage when performing calibration? 
In my case, I could raise or lower the voltage based on C3 not C4..

**Comment by faiseur-cz** [comment]
Hi Joe, or anyone else reading this :) 
First of, thanks for this project!!
I am trying to make my own version of the pcbs to learn kicad - save costs on a personal project where space isnt an issue so i could have only one pcb manufactured.

So i'm wandering if i could use directly a MCP4728 chip, not the breakout pcb?

Im far from expert!
Thanks!
Flo

**Comment by joeSeggiola** [comment]
Hi @faiseur-cz! Yes, if you're planning to re-do the board layout, you can easily switch the MCP4728 breakout with the actual chip; I didn't do it myself because I'm not good at soldering SMD. It'll work the same, just keep in mind that there are some additional components on the breakout board that you should add yourself when using the chip directly (decoupling caps and pull-up/down resistors). Fortunately there's a schematic for that, I posted it last year here:

> A different option, if you’re comfortable soldering SMD, could be make a compatible breakout board yourself, using bare SMT MCP4728. Adafruit publishes a schematic for theirs [here](https://learn.adafruit.com/assets/85938).

**Comment by faiseur-cz** [comment]
Hej! Thanks for quick answer!

Yes i was actually wandering if those "extra" components were needed..
I will see then, maybe it'd be easier and cheaper to buy a ready made breakout pcb.

Im planning to include your design for a quad DS8 project, i'll send you some infos when its done if you are interested!

Thanks again!!

Flo

**Comment by joeSeggiola** [comment]
@faiseur-cz Pull-up/down resistors are needed for sure. Coupling caps may be optional if you have a stable power supply, but I can't say for sure: try searching for info in the MCP4728 datasheet. You can surely omit the power-on LED obviously :)

**Comment by joeSeggiola** [comment]
@faiseur-cz And yeah, sure, keep me posted! It's always nice to get feedback. You can find my email on my GitHub profile page, if you want.
